### PR TITLE
Support camera poses

### PR DESCRIPTION
**Pull request by slimbuck**
Fixes: #186

https://github.com/user-attachments/assets/b09bc3d2-d3e1-41f1-8ece-3ae3e69d27c7

This PR includes the following changes:
* add the ability to load colmap `cameras.json` format camera poses
* displays the list of poses in a list on the view options panel
    * the list will likely be moved to its own popup window
* supports switching between camera poses and playing back the poses as an animation
* update input handling for popup windowsb
* update npm dependencies except eslint
* fix spotty grid rendering

### CODE REVIEW

**Comment by willeastcott** [comment]
Cool. But I still strongly feel that this should be its own panel.

File: rollup.config.mjs [hunk 1]
```diff
@@ -29,11 +28,28 @@ const ENGINE_PATH = path.resolve(ENGINE_DIR, 'build', ENGINE_NAME);
...
-    playcanvas: ENGINE_PATH,
-    pcui: PCUI_DIR
+const outputHeader = () => {
+    const BLUE_OUT = '\x1b[34m';
+    const BOLD_OUT = `\x1b[1m`;
+    const REGULAR_OUT = `\x1b[22m`;
+    const RESET_OUT = `\x1b[0m`;
+
+    const title = [
+        `Building PlayCanvas Model Viewer`,
```
Comment (willeastcott) [inline rollup.config.mjs:38]: Wrong app

File: src/file-handler.ts [hunk 2]
```diff
@@ -86,9 +88,56 @@ const writeToFile = async (stream: FileSystemWritableFileStream, data: ArrayBuff
     await stream.close();
 };
 
+const loadCameraPoses = (url: string, filename: string, events: Events) => {
+    return fetch(url)
+    .then((response) => response.json())
+    .then((json) => {
```
Comment (willeastcott) [inline src/file-handler.ts:94]: Don't you prefer the async/await form? 😄

File: src/ui/view-panel.ts [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { Events } from '../events';
 import { Tooltips } from './tooltips';
 import { localize } from './localization';
 
+type Vec3d = { x: number, y: number, z: number };
```
Comment (willeastcott) [inline src/ui/view-panel.ts:6]: Isn't this just a PlayCanvas Vec3?

**Review by willeastcott — APPROVED** [review]

File: src/file-handler.ts (hunk 2 above)
Comment (slimbuck) [reply src/file-handler.ts]: I do. And since I am a modern man I have updated the code. Thanks!

File: src/ui/view-panel.ts (hunk 3 above)
Comment (slimbuck) [reply src/ui/view-panel.ts:6]: Yep might as well be. Thanks

**Pull request merged**